importexport/musicxml/importmxmlpass2.cpp

```diff
@@ -2442,6 +2443,20 @@ static void preventNegativeTick(const Fraction& tick, Fraction& offset, MxmlLogg
             }
       }
 
+//---------------------------------------------------------
+//   isTempoOrphanCandidate
+//---------------------------------------------------------
+
+bool MusicXMLDelayedDirectionElement::isTempoOrphanCandidate() const
+      {
+      return _element->isStaffText()
+            && _placement == "above"
+            && toStaffText(_element)->xmlText().contains("<b>");
```

Comment:
seems a fragile way of determining whether text might be intended as a tempo marking - esp. the `contains("<b>")` bit, even if it were true that using bold styling was the right measure to use, wouldn't it be better to explicitly query the formatting information to check for bold? And will this work if the bold comes from the Style?

Reply:
Indeed this is a fairly wide net for candidates; however, it only actually does anything if there is an existing tempo marking at the exact same staff and tick. This does risk finding a false positive if there is a score that happens to have (inexplicably) bolded direction text on the same beat as a tempo marking (and above the staff). This case is likely quite rare (bolded text above the staff is almost always a tempo direction), but nonetheless we may put this behind an importing option when porting this to master. 

As far as the `contains("<b>")`, it's true that it's not the prettiest, but MusicXML doesn't have styles as such. All the formatting is part of the <text> element, which gets translated to MuseScore's text XML (i.e. a  `font-weight="bold"` gets translated to `<b>` tags) and applied to the element with `setXmlText()`. This is because sometimes a direction will include multiple different text elements, with some portions bold and some not. For these reasons, directly checking the style of the element is always false, so checking the XML for `<b>` is most closely correlated with the MusicXML import. I'll change it from `contains("<b>")` to a simple regex to make it a bit more robust.

Reply:
Yes, I understand how xmlTest() works currently, and it's likely that in its current form checking for `<b>` is reliable enough, but in future it mightn't work if there's a switch to using, say,  `<strong>` or `<b style="font-size:large">` etc. etc
Did you consider checking it at the point of import from the MusicXML format (which is defined by a standard that's unlikely to change any time soon) ? Understand that would be a bigger change but just seems more robust.

Reply:
From a future-proofing standpoint that would make sense. I'm a bit wary that the complexity added in checking it at the point of import will itself be rigid/error-prone (partly because the existing direction import classes—`MusicXMLParserDirection` and `MusicXMLDelayedDirectionElement`—could use a good refactor). Also, the regression tests I've added would in theory catch the error if such a refactor were to occur, and certainly one would have to be checking the regression tests actively if they were to make a refactor so big.

...but I'll give it a try!

Reply:
Done! A little cumbersome, since `isBold` has to be passed down from `MusicXMLParserDirection` to `MusicXMLDelayedDirectionElement`, but I agree that this is more future-proof.